src/network/vlan.rs

```diff
@@ -191,7 +225,7 @@ fn setup(
                         // if last element return directly
                         if i == 2 {
                             return Err(NetavarkError::msg(format!(
-                                "create macvlan interface: {}",
+                                "create vlan interface: {}",
```

Comment:
I don't like the error message changes, it should be clear what driver failed for a user and for me when I read the bug reports. `vlan` is already used with a different meaning in networking so it might confuse users.

I think you can fix this by replacing the vlan part with `{}` and then just pass the kind enum as argument. That would require you to implement the Display trait which should just print the name, see https://doc.rust-lang.org/std/fmt/trait.Display.html

---
This comment applies to all changes were your changed the error message obviously.

Reply:
good idea, done